src/main/java/core/basesyntax/shopimpl/service/IOdataFileService.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax.shopimpl.service;
+
+import core.basesyntax.model.shopstrategy.ShopTransactionsType;
+import core.basesyntax.shopimpl.entity.DataRecord;
+import core.basesyntax.shopimpl.entity.Fruit;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IOdataFileService {
```

Comment:
SRP tells us that we have to split writing and reading into different modules (classes, for example)